x-pack/solutions/security/plugins/elastic_assistant/server/lib/attack_discovery/evaluation/helpers/get_graph_input_overrides/index.ts

```diff
@@ -5,25 +5,27 @@
  * 2.0.
  */
 
-import { pick } from 'lodash/fp';
-
+import type { AttackDiscoveryGraphState } from '../../../../langchain/graphs';
 import { ExampleInputWithOverrides } from '../../example_input';
-import { GraphState } from '../../../graphs/default_attack_discovery_graph/types';
 
 /**
  * Parses input from an LangSmith dataset example to get the graph input overrides
  */
-export const getGraphInputOverrides = (outputs: unknown): Partial<GraphState> => {
+export const getGraphInputOverrides = (outputs: unknown): Partial<AttackDiscoveryGraphState> => {
   const validatedInput = ExampleInputWithOverrides.safeParse(outputs).data ?? {}; // safeParse removes unknown properties
 
-  const { overrides } = validatedInput;
+  const { replacements, overrides } = validatedInput;
+
+  // Fallback to and rename the root level legacy properties `anonymizedAlerts`,
+  // and `attackDiscoveries` to the new graph property names:
+  const anonymizedDocuments = validatedInput.anonymizedDocuments ?? validatedInput.anonymizedAlerts;
+  const insights = validatedInput.insights ?? validatedInput.attackDiscoveries;
```

Comment:
This fixes an issue where Attack discovery evaluations returned lower-than-expected scores because the legacy Langsmith dataset examples were not overriding the new graph state properties. Addressed in https://github.com/elastic/kibana/pull/216757/commits/48ded9315a027bc5a7eed2a1eb07741cb976af98